src/identity.js

```diff
@@ -1663,106 +1622,63 @@ export default function Identity(mpInstance) {
-                    );
+                    self.reinitForwardersOnUserChange(prevUser, newUser);
+                    self.setForwarderCallbacks(newUser, method);
                 }
-                var newIdentitiesByName = {};
 
-                for (var key in newIdentitiesByType) {
-                    newIdentitiesByName[
-                        Types.IdentityType.getIdentityName(
-                            mpInstance._Helpers.parseNumber(key)
-                        )
-                    ] = newIdentitiesByType[key];
-                }
+                const newIdentitiesByName = getNewIdentitiesByName(
+                    newIdentitiesByType
+                );
+
+                const uiByName =
+                    method === Modify ? previousUIByName : incomingMpidUIByName;
 
                 self.sendUserIdentityChangeEvent(
                     newIdentitiesByName,
-                    method,
                     identityApiResult.mpid,
-                    method === Modify
-                        ? previousUIByNameCopy
-                        : incomingMpidUIByNameCopy
+                    uiByName
                 );
             }
 
             if (callback) {
-                if (xhr.status === 0) {
-                    mpInstance._Helpers.invokeCallback(
-                        callback,
-                        HTTPCodes.noHttpCoverage,
-                        identityApiResult || null,
-                        newUser
-                    );
-                } else {
-                    mpInstance._Helpers.invokeCallback(
-                        callback,
-                        xhr.status,
-                        identityApiResult || null,
-                        newUser
-                    );
-                }
-            } else {
-                if (
-                    identityApiResult &&
-                    identityApiResult.errors &&
-                    identityApiResult.errors.length
-                ) {
-                    mpInstance.Logger.error(
-                        'Received HTTP response code of ' +
-                            xhr.status +
-                            ' - ' +
-                            identityApiResult.errors[0].message
-                    );
-                }
+                const callbackCode =
+                    xhr.status === 0 ? HTTPCodes.noHttpCoverage : xhr.status;
+
+                mpInstance._Helpers.invokeCallback(
+                    callback,
+                    callbackCode,
+                    identityApiResult || null,
+                    newUser
+                );
+            } else if (
+                identityApiResult &&
+                !isEmpty(identityApiResult.errors)
+            ) {
+                mpInstance.Logger.error(
+                    'Received HTTP response code of ' +
+                        xhr.status +
+                        ' - ' +
+                        identityApiResult.errors[0].message
```

Comment:
Can we add a go link for figuring out why we'd only want to return the 1st error here?  It could start with figuring out what potential errors would be returned.